File: generated-code/codecraft/javascript/TcpReadWrite/model/vec2-int.js
/**
 * 2 dimensional vector.
 */
class Vec2Int {
    /**
     * `x` coordinate of the vector
     */
    x;
    /**
     * `y` coordinate of the vector
     */
    y;

    constructor(x, y) {
        this.x = x;
        this.y = y;
    }

    /**
     * Read Vec2Int from input stream
     */
    static async readFrom(stream) {
        let x;
        x = await stream.readInt();
        let y;
        y = await stream.readInt();
        return new Vec2Int(x, y);
    }

    /**
     * Write Vec2Int to output stream
     */
    async writeTo(stream) {
        let x = this.x;
        await stream.writeInt(x);
        let y = this.y;
        await stream.writeInt(y);
    }
}
module.exports = Vec2Int

File: generated-code/codecraft/scala/TcpReadWrite/src/main/scala/model/Vec2Int.scala
package model

import util.StreamUtil

/**
 * 2 dimensional vector.
 *
 * @param x `x` coordinate of the vector
 * @param y `y` coordinate of the vector
 */
case class Vec2Int(x: Int, y: Int) {
    /**
     * Write Vec2Int to output stream
     */
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, x)
        StreamUtil.writeInt(stream, y)
    }

    /**
     * Get string representation of Vec2Int
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("Vec2Int { ")
        stringBuilder.append("x: ")
        stringBuilder.append(x)
        stringBuilder.append(", ")
        stringBuilder.append("y: ")
        stringBuilder.append(y)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object Vec2Int {
    /**
     * Read Vec2Int from input stream
     */
    def readFrom(stream: java.io.InputStream): Vec2Int = Vec2Int(
        StreamUtil.readInt(stream),
        StreamUtil.readInt(stream)
    )
}